### ISSUE TITLE
Extra arguments apart from what is declared, are not attached correctly

### ISSUE THREAD

**Issue by smac89** [nat-n/poethepoet#69]
I just want to add a disclaimer: _I'm not sure whether `poe` was designed the accept extra arguments beyond what was declared for a task, but I noticed that supplying extra args after `--` seems to include the extra arguments, so that's what I was trying here_

For example, given the following task declaration:

```toml
[tool.poe.env]
PYTHONPATH = "${PWD}/examples"
DJANGO_SETTINGS_MODULE = "config.settings"

[tool.poe.tasks.migrate]
cmd = "django-admin migrate --pythonpath ${PWD}/examples/$example"
args = [{name = "example", default = "my_app_example", positional = true}]
envfile = ".env"
```

If I run it using `poe migrate -- --help`, poe will translate it to:

```sh
django-admin migrate --pythonpath /home/username/projects/my_django_app/examples/--help
```

This causes an error because as you can see, there is no space between the last argument and `--help`, however without the extra `-- --help`, everything works, but I can't pass more arguments to the command.

Please help me know if this is defined behavior and how to work around it, or whether it's a bug. Thanks

**Comment by nat-n** [comment]
Hi @smac89, thanks for the feedback!

The CLI argument parser does not give any special meaning to `--`, it simple passes it to argparse like another argument. Now it appears to be an "undocumented feature" of the argparse module that anything following `--` in the list of arguments is taken as positional argument. In the context of how argparse is used this follows POSIX convention.

So what I believe is happening in your example is that `--help` is actually being parsed as the `example` positional argument.

Relevant code is [around here](https://github.com/nat-n/poethepoet/blob/main/poethepoet/task/args.py#L237).

As for the question of what *should* poe do with `--` in the arguments list? Earlier I considered using it for invoking multiple tasks in one line, but wasn't sure it would work well. Maybe passing all remaining arguments to the command as you say would make sense, a bit like how npm does it.

I need to sleep on it. I'm open to arguments one way or another about how to best handle this.

**Comment by nat-n** [comment]
This was shipped in [0.19.0](https://pypi.org/project/poethepoet/0.19.0/) 🙂
